structuralcodes/sections/_generic.py

```diff
@@ -478,6 +581,233 @@ def calculate_moment_curvature(
 
         return res
 
+    def calculate_interaction_domain(
+        self, theta: float = 0, n_axial: int = 100
+    ) -> s_res.NmInteractionDomain:
+        """Calculate the NM interaction domain.
+
+        Arguments:
+        theta (float, default = 0): inclination of n.a. respect to y axis
+        n_axial (int, default = 100): number of discretization for axial load
+
+        Return:
+        (NmInteractionDomain)
+        """
+        # Prepare the results
+        res = s_res.NmInteractionDomain()
+        res.theta = theta
+        res.n_axial = n_axial
+        # Create an array of axial loads
+        res.n = np.linspace(self.n_min, self.n_max, n_axial)
+        # Initialize the result's arrays
+        res.m_y = np.zeros_like(res.n)
+        res.m_z = np.zeros_like(res.n)
+        # Compute strength for given angle of NA
+        for i, n_i in enumerate(res.n):
+            res_bend_strength = self.calculate_bending_strength(
+                theta=theta, n=n_i
+            )
+            res.m_y[i] = res_bend_strength.m_y
+            res.m_z[i] = res_bend_strength.m_z
+
+        return res
+
+    def calculate_interaction_domain_2(
```

Comment:
Please make this implementation our preferred one, and consider the method name and arg name above 😃

Reply:
We can delete the slow version if you want.. For now I am renaming the _2 as the default one and the old one as _old

Reply:
Good 😃 yes, please remove the old one.